### ISSUE TITLE
Error when using same partial twice in a row

### ISSUE THREAD

**Issue by hobofan** [sunng87/handlebars-rust#143]
I get the following erro when using the same partial two times in a row:
```
Error rendering "template" line 3, col 1: Cannot include self in >
```

reduced code:
```rust
let main_template = "/home/hobofan/tmp/testcase/main.hbs";
let two_partial = "/home/hobofan/tmp/testcase/two.partial.hbs";

handlebars.register_template_file("template", &Path::new(main_template)).ok().unwrap();
handlebars.register_template_file("two", &Path::new(two_partial)).ok().unwrap();

let data = BTreeMap::<String, String>::new();

handlebars.render("template", &data).unwrap_or_else(|e| format!("{}", e))
```

`main.hbs`:
```
one
{{> two }}
three
{{> two }}
```

`two.partial.hbs`:
```
----- TWO ----
```

------

I am not 100% sure, but I think that should be valid Handlebars. When I use another partial between the two uses of `{{> two }}`, everything works fine.

I'm happy to provide any further information if that helps.

**Comment by sunng87** [comment]
Thank you for reporting. It seems to be a bug for checking template name. I will look into it soon.

**Comment by sunng87** [comment]
Released in 0.25.2

**Comment by hobofan** [comment]
Thanks for the quick fix! :+1:
